frontend/public/components/utils/name-value-editor.jsx

```diff
@@ -89,17 +88,21 @@ export const NameValueEditor = withDragDropContext(
         secrets,
         addConfigMapSecret,
         toolTip,
+        t,
       } = this.props;
+      const nameString = this.props.nameString ? this.props.nameString : t('environment~Key');
+      const valueString = this.props.valueString ? this.props.valueString : t('environment~Value');
       const pairElems = nameValuePairs.map((pair, i) => {
         const key = _.get(pair, [NameValueEditorPair.Index], i);
         const isEmpty = nameValuePairs.length === 1 && nameValuePairs[0].every((value) => !value);
-
         return (
           <PairElement
             onChange={this._change}
             index={i}
-            nameString={nameString}
-            valueString={valueString}
+            nameString={this.props.nameString}
+            nameKey={!this.props.nameString && t('environment~key')}
+            valueString={this.props.valueString}
+            valueKey={!this.props.valueString && t('environment~value')}
```

Comment:
I'm unclear why we need both `nameString` and `nameKey`. Can you explain a bit? It would be better to have only one if we can.

`nameKey` is kind of misleading here since it's not the translation key, it's the actual message.

Reply:
That's a good point; I didn't spend a lot of time on naming this prop. There's some minor string manipulation (lowercasing) that goes on in PairElement, and I was using this prop to avoid the string manipulation for i18n strings. If we avoid doing the string manipulation inside that component, we don't need `nameKey`.

Reply:
@rebeccaalpert Can you share what values are being lowercased? Maybe with a screenshot of the context where it's used?

Reply:
It's in the placeholders (some of which you've commented on). `placeholder={nameKey ? nameKey : nameString.toLowerCase()}` is an example. The other placeholder I looked at as part of i18n was also lowercase. Maybe it's just better to make it a `placeholder` prop?

<img width="979" alt="Screen Shot 2020-10-27 at 3 32 47 PM" src="https://user-images.githubusercontent.com/7014965/97352962-c7f09e80-1869-11eb-9e91-52f498196152.png">

<img width="564" alt="Screen Shot 2020-10-27 at 3 33 12 PM" src="https://user-images.githubusercontent.com/7014965/97352971-ccb55280-1869-11eb-9086-086911a17a31.png">

Reply:
If it's only used for a placeholder, yeah, I think it's better to have "placeholder" somewhere in the name. I'm not sure we need to lowercase the values, and probably don't want to since we don't know what the translated string will be.